src/services/user.ts

```diff
@@ -0,0 +1,61 @@
+import mongoose from 'mongoose';
+import { Request } from 'express';
+import { IUser } from '../schemas/user/type';
+
+const User = mongoose.model('User');
+
+class UserService {
+  login(login: IUser['login'], password: IUser['password']) {
+    /*
+     * In real need to check login
+     * and compare hashed password and set token
+     * in the cookies or in the header
+     * for now just for example
+     * */
+    return User.findOne({ login, password });
```

Comment:
you need to await operations with DB here and in some cases below

Reply:
Sure for more readability, I will change. but actually, it will works right. It returns async func and in the handler, I am always awaiting for result
 
```async login(req: Request, res: Response, next: NextFunction) {
    try {
      const data = await userService.login(req.body.login, req.body.password);
      res.status(200).send(data);
    } catch (err) {
      next(err);
    }
  }
  ```